RecoLocalCalo/HcalRecAlgos/python/hcalPulseShapesESProd_cfi.py

```diff
@@ -0,0 +1,12847 @@
+import FWCore.ParameterSet.Config as cms
+
+hcalPulseShapesESProd = cms.ESProducer('HcalPulseShapesEP',
+    productLabel = cms.string('HcalDataShapes'),
+    pulseDumpFile = cms.untracked.string(''),
+    mapDumpFile = cms.untracked.string(''),
+    pulseShapeLength = cms.uint32(250),
+    globalTimeShift = cms.int32(-75),
+    defaultTimeShift = cms.int32(0),
+    pulseShapes = cms.VPSet(
+        cms.PSet(
+            label = cms.string('default'),
+            t0 = cms.uint32(0),
+            pulse = cms.vdouble(
+                5.22174e-12, 7.04852e-10, 3.49584e-08, 7.78029e-07, 9.11847e-06,
```

Comment:
wouldn't this large amount of parameters be better stored in the database, or at least in some external `cms-data` repository? This makes this file ~ unreadable.

Reply:
Yes, this was discussed with the Hcal calibration group. However, this is the first time we are introducing this channel-by-channel dependence since the beginning of CMS operations, and the pulse shapes are not expected to change more often than once per era, so it was decided to keep the information in a config file. While this file is large (and, of course, computer-generated), its structure is rather simple, as can be seen from the "fillDescriptions" function of the HcalPulseShapesEP.cc code.

Reply:
This approach is not acceptable.

Please move the pulse shapes either to a text file in an external `cms-data` package, or to a database payload.

Reply:
assign core

Reply:
assign alca

Reply:
Could you please substantiate your statement? Haven't seen anything against this approach in CMSSW guidelines and can't understand why it can be harmful.

Reply:
Have you seen many other modules with a configuration that takes **twelve thousand lines** of python code ?